STT/whisper_stt_handler.py

```diff
@@ -100,11 +117,24 @@ def process(self, spoken_prompt):
 
         input_features = self.prepare_model_inputs(spoken_prompt)
         pred_ids = self.model.generate(input_features, **self.gen_kwargs)
+        language_id = self.processor.tokenizer.decode(pred_ids[0, 1])
+
+        if language_id not in SUPPORTED_LANGUAGES:  # reprocess with the last language
+            logger.warning("Whisper detected unsupported language:", language_id)
+            gen_kwargs = copy(self.gen_kwargs)
+            gen_kwargs['language'] = self.last_language
+            language_id = self.last_language
+            pred_ids = self.model.generate(input_features, **gen_kwargs)
+        else:
+            self.last_language = language_id
+        
```

Comment:
Not sure to get the logic here. From what I understand, when the detected language is not supported by what's coming after (namely MeloTTS), we re-transcribe using the previously detected language. Even if in practice it tends to work (if you have a french sentence and transcribe by setting language to spanish it will actually translate), this is not an expected behavior for Whisper and might lead to hallucinations (see [here](https://github.com/openai/whisper?tab=readme-ov-file#approach) for expected usage). Maybe it would be better to prompt the LLM with `"Repeat: The language you have used is not supported. Please talk in one of the expected languages"` and set language_code to `en` in this case.